sensor/tests/resource/helper.go

```diff
@@ -413,10 +431,49 @@ func (c *TestContext) ApplyFile(ctx context.Context, ns string, file YamlTestFil
 	}
 
 	return func() error {
+		if isAwaitedResource(file.Kind) {
+			err := c.r.Delete(ctx, obj)
+			if err != nil {
+				return err
+			}
+
+			// wait for deletion to be finished
+			if err := wait.For(conditions.New(c.r).ResourceDeleted(obj)); err != nil {
+				fmt.Println("failed to wait for resource deletion")
+			}
+			return nil
+		}
 		return c.r.Delete(ctx, obj)
 	}, nil
 }
 
+type waitFunc func(time.Duration) error
+
+func createResourceFunc(ctx context.Context, r *resources.Resources, obj k8s.Object) func(time.Duration) error {
+	return func(sleepTime time.Duration) error {
+		if err := r.Create(ctx, obj); err != nil {
+			time.Sleep(sleepTime)
+			return err
+		}
+		return nil
+	}
+}
+
+func execWithRetry(timeout time.Duration, fn waitFunc) error {
```

Comment:
1. Makes sense, I was supposed to rename the function to `execFunc` or something like that. 
2. Good catch

I didn't know about that backoff library. Everything makes sense; I've changed the code to use it, but I kept the parameters to the `execWithRetry` function just in case we want to use it with different timeouts and intervals. Let me know what you think.